ras-aer-handler.c

```diff
@@ -52,6 +57,86 @@ static const char *aer_uncor_errors[32] = {
 	[20] = "Unsupported Request",
 };
 
+#ifdef HAVE_AMP_NS_DECODE
+#define IPMITOOL_CMD "/usr/bin/ipmitool"
+#define DMIDECODE_CMD "/usr/sbin/dmidecode"
+static bool ampere_ipmitool = false;
+
+static void ras_report_aer_ipmi_init(void)
+{
+	struct utsname unm;
+	struct stat st;
+	int rc;
+
+	/*
+	 * Verify on startup if we are on an Ampere Altra or Altra Max
+	 * platform, to set the use of ipmitool (if installed).
+	 */
+	if (stat(IPMITOOL_CMD, &st) != 0)
+		return;
+
+	if ((uname(&unm) != 0) || (strncmp(unm.machine, "aarch64", 8) != 0))
+		return;
+
+	/* prefer dmidecode as lscpu may not have the necessary dmi info */
+	if (stat(DMIDECODE_CMD, &st) == 0)
+		rc = system(DMIDECODE_CMD" -t 4 | /usr/bin/grep "
+		    "'Ampere(R) Altra(R)' > /dev/null");
+	else
+		rc = system("/usr/bin/lscpu | /usr/bin/grep "
+		    "'Ampere(R) Altra(R)' > /dev/null");
+	if (rc == -1 || !WIFEXITED(rc) || WEXITSTATUS(rc))
```

Comment:
Instead of doing that, just open the relevant entry(ies) from /sys/class/dmi/id/ and/or /proc/cpuinfo. That won't require having hardcoded patches there for grep/lscpu/dmidecode (which would require some way to be configured, as the tool location may vary between different distros, so it would require some logic at configure.ac in order to allow customizing it).

Reply:
v2: Agreed !
Unfortunately there is no simple kernel entry giving this information, as it is a BIOS data provided in the DMI tables.
So I adapted a piece of code from the lscpu system command decoding the part of DMI tables to retrieve the 'modelname' information needed to identify the 'Ampere(R) Altra(R)' platform.
(Same GPL-2.0 license as rasdaemon)